paasta_tools/cli/cmds/logs.py

```diff
@@ -559,6 +610,86 @@ def cluster_to_scribe_env(self, cluster):
+    """
+    Parses the --from and --to arguments to create python datetime objects representing
+    the start and end times for log retrieval
+    """
+    if to_string is None:
+        end_time = datetime.datetime.utcnow()
+    else:
+        # Try parsing as a a natural time duration first, if that fails move on to
+        # parsing as an ISO-8601 timestamp
+        to_duration = timeparse(to_string)
+
+        if to_duration is not None:
+            end_time = datetime.datetime.utcnow() - datetime.timedelta(seconds=to_duration)
+        else:
+            end_time = isodate.parse_datetime(to_string)
+
+    from_duration = timeparse(from_string)
+    if from_duration is not None:
+        start_time = datetime.datetime.utcnow() - datetime.timedelta(seconds=from_duration)
+    else:
+        start_time = isodate.parse_datetime(from_string)
+
+    # Covert the timestamps to something timezone aware
+    start_time = pytz.utc.localize(start_time)
+    end_time = pytz.utc.localize(end_time)
+
+    if start_time > end_time:
+        raise ValueError("Start time bigger than end time")
+
+    return start_time, end_time
+
+
+def validate_filtering_args(args, log_reader):
+    if not log_reader.SUPPORTS_LINE_OFFSET and args.line_offset is not None:
+        print PaastaColors.red(log_reader.__class__.__name__ + " does not support line based offsets")
+        return False
+    if not log_reader.SUPPORTS_LINE_COUNT and args.line_count is not None:
+        print PaastaColors.red(log_reader.__class__.__name__ + " does not support line count based log retrieval")
+        return False
+    if not log_reader.SUPPORTS_TAILING and args.tail:
+        print PaastaColors.red(log_reader.__class__.__name__ + " does not support tailing")
+        return False
+    if not log_reader.SUPPORTS_TIME and (args.time_from is not None or args.time_to is not None):
+        print PaastaColors.red(log_reader.__class__.__name__ + " does not support time based offsets")
+        return False
+
+    if args.tail and (args.line_count is not None or args.time_from is not None or
+                      args.time_to is not None or args.line_offset is not None):
+        print PaastaColors.red("You cannot specify line/time based filtering parameters when tailing")
+        return False
+
+    # Can't have both
+    if args.line_count is not None and args.time_from is not None:
+        print PaastaColors.red("You cannot filter based on both line counts and time")
+        return False
+
+    return True
+
+
+def pick_default_log_mode(args, log_reader, service, levels, components, clusters):
+    if log_reader.SUPPORTS_LINE_COUNT:
+        sys.stderr.write(PaastaColors.cyan("No filtering specified, grabbing last 100 lines") + "\n")
+        log_reader.print_last_n_logs(service, 100, levels, components, clusters, raw_mode=args.raw_mode)
+        return 0
+
+    elif log_reader.SUPPORTS_TIME:
+        start_time, end_time = generate_start_end_time()
```

Comment:
Not here, since this is just from the default arguments so it should never fail. Will add one for main though